eth/downloader/peer.go

```diff
@@ -305,13 +311,42 @@ func (p *peer) Demote() {
 	}
 }
 
+// Ignore appends a new entity to the set of items (blocks, receipts, states)
+// that a peer is known not to have (i.e. have been requested before). If the
+// set reaches its maximum allowed capacity, items are randomly dropped off.
+func (p *peer) Ignore(hash common.Hash) {
+	p.ignoreLock.Lock()
+	defer p.ignoreLock.Unlock()
+
+	p.ignored[hash] = struct{}{}
+	for len(p.ignored) > maxIgnoredHashes {
+		for drop, _ := range p.ignored {
+			delete(p.ignored, drop)
+			break
+		}
+	}
```

Comment:
True, but I'm not sure it's worth the complexity of adding another construct in there. It's true that this map is cimple and stupid, but as long as it works I'd rather not complicate matters.